src/Grid.jsx

```diff
@@ -1,10 +1,12 @@
 import { useState, useRef, Fragment } from "react";
-import { PatternPoint, deletePoint } from "./PatternPoint";
-import { SegmentPath } from "./SegmentPath";
-import { CurvePath } from "./CurvePath";
+import { PatternPoint, deletePoint } from "./PointLogic/PatternPoint";
```

Comment:
suggestion: the `deletePoint` function doesn't need to be defined in that other file (it's only used in `Grid.jsx`), and also could be better defined as a custom hook.